src/Hosting/TestHost/src/WebSocketClient.cs

```diff
@@ -76,6 +77,7 @@ public async Task<WebSocket> ConnectAsync(Uri uri, CancellationToken cancellatio
                 request.Headers.Add("Upgrade", new string[] { "websocket" });
                 request.Headers.Add("Sec-WebSocket-Version", new string[] { "13" });
                 request.Headers.Add("Sec-WebSocket-Key", new string[] { CreateRequestKey() });
+                request.Headers.Add("Sec-WebSocket-Protocol", SubProtocols.ToArray());
```

Comment:
This should only be called if SubProtocols is not empty.

Reply:
I'll add to one of the existing tests (or a new test) to assert the absence of the header when the list is empty.